src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/Rule/TargetGroupEvaluatorTest.php

```diff
@@ -63,16 +63,14 @@ public function setUp(): void
         );
     }
 
-    /**
-     * @dataProvider provideEvaluationData
-     */
+    #[\PHPUnit\Framework\Attributes\DataProvider('provideEvaluationData')]
     public function testEvaluate(
-        $targetGroups,
-        $ruleWhitelists,
-        $webspaceKey,
-        $evaluatedTargetGroup,
-        $frequency = TargetGroupRuleInterface::FREQUENCY_SESSION,
-        $currentTargetGroup = null
+        array $targetGroups,
+        array $ruleWhitelists,
+        ?string $webspaceKey,
+        ?TargetGroup $evaluatedTargetGroup,
+        int $frequency = TargetGroupRuleInterface::FREQUENCY_SESSION,
+        ?TargetGroup $currentTargetGroup = null
```

Comment:
test method changes should be done on 2.5 branch if possible and non breaking changes